### DISCUSSION TITLE
How to set up compiling for modern Android?

### DISCUSSION THREAD

**Discussion by John-Dekka** [HaxeFlixel/flixel#2633] [General]
Hy there,

I'd like to compile to Android, but all tutorials I can find are quite outdated and use old Android SDK and NDK versions. Lime tries to compile using gcc, which was replaced with clang in the sdk quite a while ago.

I am using latest haxeflixel in an vscode environment on a Fedora workstation.

**Comment by Geokureli** [comment]
I don't know how many Flixel contributers and users who lurk this discussions page are compiling to Android, unfortunately. I would love to update out-of-date documentation but I'm not very familiar, myself. 

I would see if lime or openfl has more up to date documentation on this. If not, try asking in the [OpenFL Forum](https://community.openfl.org/)

**Comment by jrogers-dev** [comment]
Attempting to build for Android and also running into the issue of not being able to find "arm-linux-androideabi-g++". I have some limited experience with Android development as a whole, and I'm like a dog with a bone when I run into issues like this. So far, I've found at least one solution, which is to build the toolchain yourself, but I'd like to find a better one since building toolchains is deprecated. Will update on what I find.

**Comment by jrogers-dev** [comment]
Alright, found a solution. But, also a recommendation that you probably shouldn't do this.

Since GCC was abandoned with Android NDK 16b, I installed NDK 15c, which can be found here:
https://github.com/android/ndk/wiki/Unsupported-Downloads

Simply extract that zip file and run "lime setup android", then point to that NDK instead of the one installed by Android Studio.
So far, I have it compiling with no warnings... yet.

However, while this *may* end up running in an emulator and *may* run on a device if the apk is sideloaded, you will NOT be able to use any of the modern Android amenities, such as the Play Store and everything that comes with that. Games on the Play Store require an API target of 30 (HaxeFlixel uses 19)

If anyone would like to implement AGDK (The Android Game Development Kit) into HaxeFlixel, that would be the way to modernize building for Android. Those interested at taking a swing at it can start here:
https://developer.android.com/games/develop/custom/overview

Unfortunately, until then, you have to use one of the recommended engines: Defold, Godot, Unreal or Unity.

You can learn more about the AGDK here:
https://developer.android.com/games

And more about the individual engines here:
https://defold.com/
https://godotengine.org/
https://www.unrealengine.com/en-US
https://unity.com/

**Reply by jrogers-dev** [reply]
Also, until someone does implement the AGDK into HaxeFlixel, the website and documentation should NOT be advertising Android as one of the cross-platform targets.
